sessiontxn/isolation/base.go

```diff
@@ -256,6 +257,16 @@ func (p *baseTxnContextProvider) ActivateTxn() (kv.Transaction, error) {
 	if p.enterNewTxnType == sessiontxn.EnterNewTxnBeforeStmt && !sessVars.IsAutocommit() && sessVars.SnapshotTS == 0 {
 		sessVars.SetInTxn(true)
 	}
+	if p.enterNewTxnType == sessiontxn.EnterNewTxnWithBeginStmt {
+		sessVars.SetInTxn(true)
+	}
+	if !core.IsAutoCommitTxn(p.sctx) && variable.EnableMDL.Load() {
+		p.infoSchema = infoschema.AttachMDLTableInfoSchema(p.infoSchema)
+		sessVars.TxnCtx.InfoSchema = p.infoSchema
+	}
+	if variable.EnableMDL.Load() {
+		sessVars.TxnCtx.EnableMDL = true
+	}
```

Comment:
It is better to unify these two `variable.EnableMDL.Load()`. They can be different(although the probability is low).